specs/tldr-lint.spec.js

```diff
@@ -3,185 +3,70 @@
-    expect(errors.length).toBe(1);
-  });
-
-  it('TLDR017\t' + linter.ERRORS.TLDR017, function() {
-    var errors = lintFile('pages/failing/017.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR017')).toBeTruthy();
-    expect(errors.length).toBe(1);
-  });
-
-  it('TLDR018\t' + linter.ERRORS.TLDR018, function() {
-    var errors = lintFile('pages/failing/018.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR018')).toBeTruthy();
-    expect(errors.length).toBe(2);
-  });
-
-  it('TLDR019\t' + linter.ERRORS.TLDR019, function() {
-    var errors = lintFile('pages/failing/019.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR019')).toBeTruthy();
-    expect(errors.length).toBe(1);
-  });
+describe('TLDR conventions', function() {
+  parameterizedTests([
+    ['001'],
+    ['002', 3],
+    ['003'],
+    ['004', 4, ['TLDR004', 'TLDR014']],
+    ['005', 2],
+    ['006'],
+    ['007', 2],
+    ['008'],
+    ['009'],
+    ['010', 6],
+    ['011', 2],
+    ['012', 2],
+    ['013'],
+    ['014', 5],
+    ['015'],
+    ['016'],
+    ['017'],
+    ['018', 2],
+    ['019', 1],
+  ]);
 });
 
 describe('Common TLDR formatting errors', function() {
-  it('TLDR101\t' + linter.ERRORS.TLDR101, function() {
-    var errors = lintFile('pages/failing/101.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR101')).toBeTruthy();
-    expect(errors.length).toBe(1);
-  });
-
-  it('TLDR102\t' + linter.ERRORS.TLDR102, function() {
-    var errors = lintFile('pages/failing/102.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR102')).toBeTruthy();
-    expect(errors.length).toBe(1);
-  });
-
-  it('TLDR103\t' + linter.ERRORS.TLDR103, function() {
-    var errors = lintFile('pages/failing/103.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR103')).toBeTruthy();
-    expect(errors.length).toBe(2);
-  });
-
-  it('TLDR104\t' + linter.ERRORS.TLDR104, function() {
-    var errors = lintFile('pages/failing/104.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR104')).toBeTruthy();
-    expect(errors.length).toBe(2);
-  });
-
-  it('TLDR105\t' + linter.ERRORS.TLDR105, function() {
-    var errors = lintFile('pages/failing/105.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR105')).toBeTruthy();
-    expect(errors.length).toBe(2);
-  });
-
-  it('TLDR106\t' + linter.ERRORS.TLDR106, function() {
-    var errors = lintFile('pages/failing/106.md').errors;
-    expect(containsOnlyErrors(errors, 'TLDR106')).toBeTruthy();
-    expect(errors.length).toBe(1);
-  });
-
-  it('TLDR107\t' + linter.ERRORS.TLDR107, function() {
-    var errors = lintFile('pages/failing/107').errors;
-    expect(containsOnlyErrors(errors, 'TLDR107')).toBeTruthy();
-    expect(errors.length).toBe(1);
-  });
+  parameterizedTests([
+    ['101'],
+    ['102'],
+    ['103', 2],
+    ['104', 2],
+    ['105', 2],
+    ['106'],
+    ['107', 1, undefined, '107'],
+    ['108', 1, undefined, '108 .md'],
+    ['109', 1, undefined, '109A.md'],
```

Comment:
You could also do this:
```suggestion
    ['107', 1, , '107'],
    ['108', 1, , '108 .md'],
    ['109', 1, , '109A.md'],
```

Reply:
I prefer the explicitness of putting `undefined`, it also doesn't lean on a JS specific thing making it easier to work with for people more used to other languages.